src/Microsoft.TestPlatform.CoreUtilities/ValidateArg.cs

```diff
@@ -34,7 +34,7 @@ public static class ValidateArg
     /// Type of argument.
     /// </returns>
     [DebuggerStepThrough]
-    public static T NotNull<T>([ValidatedNotNull] T? arg, string parameterName)
+    public static T NotNull<T>([NotNull] T? arg, string parameterName)
```

Comment:
What is the impact of this change?

Reply:
No impact. That also works on net7.

Reply:
I was looking for motivation of this change. Why is this better than the attribute we had?

Reply:
It's shipped inside the framework starting with .net 5 or 6 while the other one has to be manually defined.

Reply:
Attribute we had didn't suppress the warnings and we already had this so in the cases it suppressed it was just a duplicate.